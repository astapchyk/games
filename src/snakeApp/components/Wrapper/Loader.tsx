import React, { ReactElement } from 'react';

function Loader(): ReactElement {
  return <div>is loading...</div>;
}

export default Loader;
