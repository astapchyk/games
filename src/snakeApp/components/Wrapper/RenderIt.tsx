import React, { ReactElement } from 'react';

type RenderItProps = {
  conditional?: boolean;
  content: ReactElement;
};

export default function RenderIt({
  conditional,
  content,
}: RenderItProps): ReactElement {
  return <>{conditional ? content : null}</>;
}
