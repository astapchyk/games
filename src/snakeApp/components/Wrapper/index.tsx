import React, { ReactElement } from 'react';

import RenderIt from './RenderIt';

type BaseWrapperProps = {
  isEmpty?: boolean;
  emptyContent?: ReactElement;
  isLoading?: boolean;
  loadingContent?: ReactElement;
  children?: ReactElement;
};

function Loader(): ReactElement {
  return <div>is loading...</div>;
}

function Wrapper({
  isEmpty,
  emptyContent,
  isLoading,
  loadingContent = <Loader />,
  children,
}: BaseWrapperProps): ReactElement {
  return (
    <>
      <RenderIt conditional={!isLoading && isEmpty} content={emptyContent} />
      <RenderIt conditional={isLoading} content={loadingContent} />
      <RenderIt conditional={!isLoading && !isEmpty} content={children} />
    </>
  );
}

export default Wrapper;
