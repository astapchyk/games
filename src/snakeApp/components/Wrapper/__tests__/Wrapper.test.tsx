import React from 'react';
import { render } from '@testing-library/react';

import Wrapper from '../index';

describe('<Wrapper />', () => {
  test('renders the empty component with default content', async () => {
    const { container } = render(<Wrapper isEmpty />);
    expect(container).toMatchSnapshot();
  });

  test('renders the empty component with a custom content', async () => {
    const contentText = 'a custom content of the empty component';

    const { container, getByText } = render(
      <Wrapper isEmpty emptyContent={<div>{contentText}</div>} />,
    );
    expect(container).toMatchSnapshot();
    expect(getByText(contentText)).toBeInTheDocument();
  });

  test('renders the loading component with default content', async () => {
    const { container } = render(<Wrapper isLoading />);
    expect(container).toMatchSnapshot();
  });

  test('renders the loader component with a custom content', async () => {
    const contentText = 'a custom content of the loader component';

    const { container } = render(
      <Wrapper isLoading loadingContent={<div> {contentText} </div>} />,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders the render it component with data', async () => {
    const { container } = render(
      <Wrapper>
        <div>children data</div>
      </Wrapper>,
    );
    expect(container).toMatchSnapshot();
  });
});
