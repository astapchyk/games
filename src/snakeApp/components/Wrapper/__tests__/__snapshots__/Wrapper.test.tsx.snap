// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> renders the empty component with a custom content 1`] = `
<div>
  <div>
    a custom content of the empty component
  </div>
</div>
`;

exports[`<Wrapper /> renders the empty component with default content 1`] = `<div />`;

exports[`<Wrapper /> renders the loader component with a custom content 1`] = `
<div>
  <div>
     
    a custom content of the loader component
     
  </div>
</div>
`;

exports[`<Wrapper /> renders the loading component with default content 1`] = `
<div>
  <div>
    is loading...
  </div>
</div>
`;

exports[`<Wrapper /> renders the render it component with data 1`] = `
<div>
  <div>
    children data
  </div>
</div>
`;
